/*
 * This file was automatically generated by EvoSuite
 * Sun Oct 09 22:20:48 GMT 2022
 */

package org.apache.bcel.classfile;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.io.BufferedInputStream;
import java.io.BufferedOutputStream;
import java.io.ByteArrayInputStream;
import java.io.DataInput;
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.EOFException;
import java.io.FileDescriptor;
import java.io.IOException;
import java.io.OutputStream;
import org.apache.bcel.classfile.Attribute;
import org.apache.bcel.classfile.Constant;
import org.apache.bcel.classfile.ConstantPool;
import org.apache.bcel.classfile.EmptyVisitor;
import org.apache.bcel.classfile.ExceptionTable;
import org.apache.bcel.classfile.Visitor;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.mock.java.io.MockFile;
import org.evosuite.runtime.mock.java.io.MockFileInputStream;
import org.evosuite.runtime.mock.java.io.MockFileOutputStream;
import org.evosuite.runtime.mock.java.io.MockPrintStream;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class ExceptionTable_ESTest extends ExceptionTable_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      int[] intArray0 = new int[1];
      ExceptionTable exceptionTable0 = new ExceptionTable(0, (-708), intArray0, (ConstantPool) null);
      Constant[] constantArray0 = new Constant[0];
      ConstantPool constantPool0 = new ConstantPool(constantArray0);
      ExceptionTable exceptionTable1 = (ExceptionTable)exceptionTable0.copy(constantPool0);
      assertNotSame(exceptionTable1, exceptionTable0);
      assertEquals(1, exceptionTable1.getNumberOfExceptions());
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(0, (-708), intArray0, (ConstantPool) null);
      EmptyVisitor emptyVisitor0 = new EmptyVisitor();
      exceptionTable0.accept(emptyVisitor0);
      assertEquals(0, exceptionTable0.getNameIndex());
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      int[] intArray0 = new int[2];
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable(0, 0, intArray0, constantPool0);
      int int0 = exceptionTable0.getNumberOfExceptions();
      assertEquals(2, int0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      int[] intArray0 = new int[3];
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable(1749, 92, intArray0, constantPool0);
      int[] intArray1 = exceptionTable0.getExceptionIndexTable();
      assertEquals(3, intArray1.length);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(3792, 3792, intArray0, (ConstantPool) null);
      exceptionTable0.setNameIndex((-2128));
      Attribute attribute0 = exceptionTable0.copy((ConstantPool) null);
      assertNotSame(attribute0, exceptionTable0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      int[] intArray0 = new int[8];
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable(0, 0, intArray0, constantPool0);
      ExceptionTable exceptionTable1 = (ExceptionTable)exceptionTable0.copy(constantPool0);
      assertEquals(8, exceptionTable1.getNumberOfExceptions());
      assertNotSame(exceptionTable1, exceptionTable0);
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(3792, 3792, intArray0, (ConstantPool) null);
      Attribute attribute0 = exceptionTable0.copy((ConstantPool) null);
      assertNotSame(attribute0, exceptionTable0);
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      int[] intArray0 = new int[7];
      Constant[] constantArray0 = new Constant[2];
      ConstantPool constantPool0 = new ConstantPool(constantArray0);
      ExceptionTable exceptionTable0 = new ExceptionTable(0, 0, intArray0, constantPool0);
      // Undeclared exception!
      try { 
        exceptionTable0.toString();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Constant pool at index 0 is null.
         //
         verifyException("org.apache.bcel.classfile.ConstantPool", e);
      }
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      int[] intArray0 = new int[8];
      Constant[] constantArray0 = new Constant[0];
      ConstantPool constantPool0 = new ConstantPool(constantArray0);
      ExceptionTable exceptionTable0 = new ExceptionTable((-486), (-486), intArray0, constantPool0);
      // Undeclared exception!
      try { 
        exceptionTable0.getExceptionNames();
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Invalid constant pool reference: 0. Constant pool size is: 0
         //
         verifyException("org.apache.bcel.classfile.ConstantPool", e);
      }
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(0, (-708), intArray0, (ConstantPool) null);
      DataOutputStream dataOutputStream0 = new DataOutputStream((OutputStream) null);
      // Undeclared exception!
      try { 
        exceptionTable0.dump(dataOutputStream0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.io.DataOutputStream", e);
      }
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      int[] intArray0 = new int[8];
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable(0, 0, intArray0, constantPool0);
      MockFileOutputStream mockFileOutputStream0 = new MockFileOutputStream("}f/S");
      DataOutputStream dataOutputStream0 = new DataOutputStream(mockFileOutputStream0);
      dataOutputStream0.close();
      try { 
        exceptionTable0.dump(dataOutputStream0);
        fail("Expecting exception: IOException");
      
      } catch(IOException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.mock.java.io.MockFileOutputStream", e);
      }
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      ExceptionTable exceptionTable0 = null;
      try {
        exceptionTable0 = new ExceptionTable(134, 0, (DataInput) null, (ConstantPool) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.bcel.classfile.ExceptionTable", e);
      }
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      FileDescriptor fileDescriptor0 = new FileDescriptor();
      MockFileInputStream mockFileInputStream0 = new MockFileInputStream(fileDescriptor0);
      BufferedInputStream bufferedInputStream0 = new BufferedInputStream(mockFileInputStream0);
      DataInputStream dataInputStream0 = new DataInputStream(bufferedInputStream0);
      ExceptionTable exceptionTable0 = null;
      try {
        exceptionTable0 = new ExceptionTable(0, 4035, dataInputStream0, constantPool0);
        fail("Expecting exception: IOException");
      
      } catch(Throwable e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.evosuite.runtime.mock.java.io.NativeMockedIO", e);
      }
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(0, (-708), intArray0, (ConstantPool) null);
      int[] intArray1 = exceptionTable0.getExceptionIndexTable();
      assertSame(intArray1, intArray0);
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      int[] intArray0 = new int[3];
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable(1749, 1749, intArray0, constantPool0);
      // Undeclared exception!
      try { 
        exceptionTable0.toString();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.bcel.classfile.ConstantPool", e);
      }
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(0, (-708), intArray0, (ConstantPool) null);
      String string0 = exceptionTable0.toString();
      assertEquals("Exceptions: ", string0);
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      int[] intArray0 = new int[3];
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable(1749, 1749, intArray0, constantPool0);
      exceptionTable0.setExceptionIndexTable(intArray0);
      assertEquals(3, exceptionTable0.getNumberOfExceptions());
  }

  @Test(timeout = 4000)
  public void test17()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(0, (-708), intArray0, (ConstantPool) null);
      exceptionTable0.setExceptionIndexTable((int[]) null);
      assertEquals(0, exceptionTable0.getNumberOfExceptions());
  }

  @Test(timeout = 4000)
  public void test18()  throws Throwable  {
      int[] intArray0 = new int[2];
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable(0, 0, intArray0, constantPool0);
      // Undeclared exception!
      try { 
        exceptionTable0.getExceptionNames();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.bcel.classfile.ConstantPool", e);
      }
  }

  @Test(timeout = 4000)
  public void test19()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(0, (-708), intArray0, (ConstantPool) null);
      String[] stringArray0 = exceptionTable0.getExceptionNames();
      assertEquals(0, stringArray0.length);
  }

  @Test(timeout = 4000)
  public void test20()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(0, (-708), intArray0, (ConstantPool) null);
      int int0 = exceptionTable0.getNumberOfExceptions();
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test21()  throws Throwable  {
      int[] intArray0 = new int[3];
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable(1749, 1749, intArray0, constantPool0);
      MockFile mockFile0 = new MockFile("2ovalI*+", "2ovalI*+");
      MockPrintStream mockPrintStream0 = new MockPrintStream(mockFile0);
      BufferedOutputStream bufferedOutputStream0 = new BufferedOutputStream(mockPrintStream0, 3);
      DataOutputStream dataOutputStream0 = new DataOutputStream(bufferedOutputStream0);
      exceptionTable0.dump(dataOutputStream0);
      assertEquals(12L, mockFile0.length());
  }

  @Test(timeout = 4000)
  public void test22()  throws Throwable  {
      byte[] byteArray0 = new byte[6];
      byteArray0[1] = (byte)1;
      ByteArrayInputStream byteArrayInputStream0 = new ByteArrayInputStream(byteArray0, (byte)1, 22);
      DataInputStream dataInputStream0 = new DataInputStream(byteArrayInputStream0);
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = null;
      try {
        exceptionTable0 = new ExceptionTable((-1269), (-3834), dataInputStream0, constantPool0);
        fail("Expecting exception: EOFException");
      
      } catch(Throwable e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.io.DataInputStream", e);
      }
  }

  @Test(timeout = 4000)
  public void test23()  throws Throwable  {
      byte[] byteArray0 = new byte[6];
      ByteArrayInputStream byteArrayInputStream0 = new ByteArrayInputStream(byteArray0, (byte)1, 22);
      DataInputStream dataInputStream0 = new DataInputStream(byteArrayInputStream0);
      ConstantPool constantPool0 = new ConstantPool((Constant[]) null);
      ExceptionTable exceptionTable0 = new ExceptionTable((-1269), (-3834), dataInputStream0, constantPool0);
      assertEquals(3, byteArrayInputStream0.available());
  }

  @Test(timeout = 4000)
  public void test24()  throws Throwable  {
      Constant[] constantArray0 = new Constant[0];
      ConstantPool constantPool0 = new ConstantPool(constantArray0);
      ExceptionTable exceptionTable0 = new ExceptionTable(2710, 2, (int[]) null, constantPool0);
      assertEquals(0, exceptionTable0.getNumberOfExceptions());
  }

  @Test(timeout = 4000)
  public void test25()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(676, 676, intArray0, (ConstantPool) null);
      // Undeclared exception!
      try { 
        exceptionTable0.accept((Visitor) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.bcel.classfile.ExceptionTable", e);
      }
  }

  @Test(timeout = 4000)
  public void test26()  throws Throwable  {
      int[] intArray0 = new int[0];
      ExceptionTable exceptionTable0 = new ExceptionTable(676, 676, intArray0, (ConstantPool) null);
      ExceptionTable exceptionTable1 = new ExceptionTable(exceptionTable0);
      assertFalse(exceptionTable1.equals((Object)exceptionTable0));
  }

  @Test(timeout = 4000)
  public void test27()  throws Throwable  {
      ExceptionTable exceptionTable0 = null;
      try {
        exceptionTable0 = new ExceptionTable((ExceptionTable) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.bcel.classfile.ExceptionTable", e);
      }
  }
}
